### ISSUE TITLE
Ready-Only checkbox state implementation

### ISSUE THREAD

**Issue by harsh096** [Sage/carbon#6672]
### Description

Checkbox component has a read-only state in DS but its not implemented in Carbon yet. We can do some logical work-around for now, but it'll be good if we can have this prop available.

### Suggested solution

Make `readOnly` prop available for checkbox

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by edleeks87** [comment]
@harpalsingh @ljemmo @tempertemper tagging you just in case there a reason not to support readOnly checkboxes here, I ask as it's hard to know if support has been intentionally omitted here or not.

**Comment by ljemmo** [comment]
@edleeks87 i think readonly checkboxes are required for things like user management or settings. They might have permission to view their access rights but not permission to change them

**Comment by DipperTheDan** [comment]
FE-6533

**Comment by tempertemper** [comment]
I don't think you can make a checkbox readonly… Doesn't work natively in HTML; it doesn't change the default visuals and still allows the checkbox to be checked, but it removes the focus state for keyboard users.

The [HTML spec for checkboxes](https://html.spec.whatwg.org/multipage/input.html#checkbox-state-(type=checkbox)) says:

> The following content attributes must not be specified and do not apply: … readonly

I guess the reasoning must be that `readonly` prevents the user from editing a field's value and still submits that value as part of the form, but the content is still interactive (though mainly just to allow things like copy/paste); checkboxes have a *state* (they have the `checked` attribute or they don't) but not a value.

So we should remove this to:

- comply with the spec (which falls under [WCAG 4.1.1](https://www.w3.org/TR/WCAG21/#parsing))
- meet user expectations

If a checkbox's state needs to be communicated but not amended, a non-form element would be most appropriate. To use Luke's example, another content type shoujld be used; maybe a `<dl>` where the `<dt>` is the name of the permission and `<dd>` is either on or off.
